konnectivity-socks5-proxy/main.go

```diff
@@ -223,8 +277,22 @@ func (d proxyResolver) Resolve(ctx context.Context, name string) (context.Contex
 		address, err := d.guestClusterResolver.resolve(ctx, name)
 		if err != nil {
 			l.Error(err, "failed to look up address from guest cluster")
+
+			if d.resolveFromManagementCluster {
+				setDNSFallbackFlagOnce.Do(func() {
+					l.Info("Fallback to management cluster resolution")
+					d.dnsFallback.set(true)
+				})
+				return ctx, nil, nil
```

Comment:
I think we still need to do a direct resolve here, no?

Reply:
we're replicating the same behavior as with cloud provider addresses, so it's ok